tools/docker-go

```diff
@@ -37,7 +37,7 @@ parse_args() {
         --module-path ) shift; GO_MODULE_PATH="${1}" ;;
         --sdk-image ) shift; SDK_IMAGE="${1}" ;;
         --go-mod-cache ) shift; GO_MOD_CACHE="${1}" ;;
-        --command ) shift; COMMAND="${@:1}" ;;
+        --command ) shift; COMMAND="${*:1}" ;;
```

Comment:
This makes the warning go away and retains current behavior, but the warning hints at an actual problem: The only safe way to use `docker-go` is to pass a single argument to `--command`. Anything else would lead to unexpected behavior, for example `docker-go --command foo bar baz` is indistinguishable from `docker-go --command foo 'bar baz'` and `docker-go --command foo --module-path bar` will both pass `--module-path bar` to `foo`, and actually modify the script's `GO_MODULE_PATH` variable.

Is it feasible to change the script's interface without too many coordinated changes? A safer interface could interpret anything after `--` as the command to execute, and keep all following arguments in an array (`COMMAND=( "$@" )`, later: `"${COMMAND[@]}"`) to retain fidelity of the originally passed-in arguments.

If that's too much for this PR or the script can't be changed for other reasons, disabling the warning could at least serve as a sign calling attention to the unexpected behavior.

Reply:
That is a very good point. I wanted to keep the changes in this PR to the bare minimum to avoid introducing too many changes at once, but I went ahead and created issue #2534 so that we can follow-up on this.

Reply:
Makes sense, it's best to keep the PR focused. Thanks!